### ISSUE TITLE
Loss is always up and down 20 when training with PASCALVOC 2012, who know how to fix?

### ISSUE THREAD

**Issue by ivyteng** [balancap/SSD-Tensorflow#159]
I  try to train SSD using PASCAL VOC 2012 with default parameters , but the loss doesn't decrease when it reach 20. Actually the loss is always up and down 30.

INFO:tensorflow:global step 50850: loss = 43.2839 (0.300 sec/step)
INFO:tensorflow:global step 50860: loss = 28.6162 (0.289 sec/step)
INFO:tensorflow:global step 50870: loss = 34.9561 (0.295 sec/step)
INFO:tensorflow:global step 50880: loss = 35.0722 (0.287 sec/step)
INFO:tensorflow:global step 50890: loss = 36.1488 (0.290 sec/step)
INFO:tensorflow:global step 50900: loss = 22.2183 (0.303 sec/step)
INFO:tensorflow:global step 50910: loss = 33.8348 (0.289 sec/step)
INFO:tensorflow:global step 50920: loss = 28.7940 (0.294 sec/step)
INFO:tensorflow:global step 50930: loss = 21.3115 (0.296 sec/step)
INFO:tensorflow:global step 50940: loss = 31.7467 (0.295 sec/step)
INFO:tensorflow:global step 50950: loss = 41.7943 (0.298 sec/step)
INFO:tensorflow:global step 50960: loss = 22.4475 (0.294 sec/step)
INFO:tensorflow:global step 50970: loss = 38.2551 (0.303 sec/step)
INFO:tensorflow:global step 50980: loss = 31.7386 (0.293 sec/step)
INFO:tensorflow:global step 50990: loss = 41.7607 (0.296 sec/step)
INFO:tensorflow:global step 51000: loss = 37.3786 (0.295 sec/step)
INFO:tensorflow:global step 51010: loss = 41.9627 (0.294 sec/step)

**Comment by ivyteng** [comment]
INFO:tensorflow:global step 50850: loss = 43.2839 (0.300 sec/step)
INFO:tensorflow:global step 50860: loss = 28.6162 (0.289 sec/step)
INFO:tensorflow:global step 50870: loss = 34.9561 (0.295 sec/step)
INFO:tensorflow:global step 50880: loss = 35.0722 (0.287 sec/step)
INFO:tensorflow:global step 50890: loss = 36.1488 (0.290 sec/step)
INFO:tensorflow:global step 50900: loss = 22.2183 (0.303 sec/step)
INFO:tensorflow:global step 50910: loss = 33.8348 (0.289 sec/step)
INFO:tensorflow:global step 50920: loss = 28.7940 (0.294 sec/step)
INFO:tensorflow:global step 50930: loss = 21.3115 (0.296 sec/step)
INFO:tensorflow:global step 50940: loss = 31.7467 (0.295 sec/step)
INFO:tensorflow:global step 50950: loss = 41.7943 (0.298 sec/step)
INFO:tensorflow:global step 50960: loss = 22.4475 (0.294 sec/step)
INFO:tensorflow:global step 50970: loss = 38.2551 (0.303 sec/step)
INFO:tensorflow:global step 50980: loss = 31.7386 (0.293 sec/step)
INFO:tensorflow:global step 50990: loss = 41.7607 (0.296 sec/step)
INFO:tensorflow:global step 51000: loss = 37.3786 (0.295 sec/step)
INFO:tensorflow:global step 51010: loss = 41.9627 (0.294 sec/step)

**Comment by xiaoluojun** [comment]
Hi，Did you find reason with problem?

**Comment by JoaoLages** [comment]
Usually that means your learning rate is too high, try to reduce it

**Comment by xiaoluojun** [comment]
Thanks!I try to set learning_rate=0.0001.but it's not fine result.
The loss is always up and down 15.

**Comment by JoaoLages** [comment]
That's strange.. The learning rate is really low already, I would try to lower it even more or there's an error in your gradient calculation

**Comment by zhihongzeng2017** [comment]
I have the similar problem.
Here is my command line: 
python train_ssd_network.py --train_dir=/workdir/SSD-Tensorflow/logs_train/ --dataset_dir=../data/tf_VOC/VOC_2007_2012_train/ --dataset_name=pascalvoc_2012 --dataset_split_name=train --model_name=ssd_300_vgg --checkpoint_path=/workdir/SSD-Tensorflow/checkpoints_train/ssd_300_vgg.ckpt --save_summaries_secs=60 --save_interval_secs=600 --weight_decay=0.0005 --optimizer=adam --learning_rate=0.001 --batch_size=32

After two days running in aws GPU p2.xlarge ec2 instance, the loss does not go down further. 
2017-11-20 20:52:04.294340: W tensorflow/stream_executor/cuda/cuda_dnn.cc:2223]
2017-11-20 20:52:04.295665: W tensorflow/stream_executor/cuda/cuda_dnn.cc:2223]
2017-11-20 20:52:04.295928: W tensorflow/stream_executor/cuda/cuda_dnn.cc:2223]
2017-11-20 20:52:05.294654: W tensorflow/stream_executor/cuda/cuda_dnn.cc:3355]
INFO:tensorflow:Recording summary at step 113778.
INFO:tensorflow:global step 113780: loss = 11.4368 (2.020 sec/step)
INFO:tensorflow:global step 113790: loss = 15.7824 (2.022 sec/step)
INFO:tensorflow:global step 113800: loss = 13.3885 (2.022 sec/step)
INFO:tensorflow:Saving checkpoint to path /workdir/SSD-Tensorflow/logs_train/model.ckpt
INFO:tensorflow:Recording summary at step 113805.
INFO:tensorflow:global step 113810: loss = 11.2704 (2.086 sec/step)
INFO:tensorflow:global step 113820: loss = 14.5088 (2.037 sec/step)
INFO:tensorflow:global step 113830: loss = 14.6238 (2.022 sec/step)
2017-11-20 20:54:04.359411: W tensorflow/stream_executor/cuda/cuda_dnn.cc:2223] 
2017-11-20 20:54:05.260177: W tensorflow/stream_executor/cuda/cuda_dnn.cc:3355] 
INFO:tensorflow:Recording summary at step 113833.
INFO:tensorflow:global step 113840: loss = 11.6869 (2.053 sec/step)
INFO:tensorflow:global step 113850: loss = 19.6293 (2.019 sec/step)
2017-11-20 20:55:04.254637: W tensorflow/stream_executor/cuda/cuda_dnn.cc:2223] 
2017-11-20 20:55:04.532945: W tensorflow/stream_executor/cuda/cuda_dnn.cc:2223] 
2017-11-20 20:55:04.533304: W tensorflow/stream_executor/cuda/cuda_dnn.cc:2223] 
2017-11-20 20:55:05.257967: W tensorflow/stream_executor/cuda/cuda_dnn.cc:3355] 
INFO:tensorflow:global step 113860: loss = 19.3938 (2.803 sec/step)

I ran evaluation on voc_2007_test. Here is the results:
INFO:tensorflow:Evaluation [4951/4952]
INFO:tensorflow:Evaluation [4952/4952]
2017-11-20 16:52:40.352404: I tensorflow/core/kernels/logging_ops.cc:79] AP_VOC07/mAP[0.72586562954673628]
2017-11-20 16:52:45.365670: I tensorflow/core/kernels/logging_ops.cc:79] AP_VOC12/mAP[0.74828734104759553]
INFO:tensorflow:Finished evaluation at 2017-11-20-16:52:46
Time spent : 2369.489 seconds.
Time spent per BATCH: 0.478 seconds.

When I run pre-trained ssd_300_vgg on the same test data, the mAP=0.83. 

Do you have any suggestion?

**Comment by muou55555** [comment]
I have the similar problem too. Loss is down 20 to 50. why?

**Comment by muou55555** [comment]
I think thd point "loss = tf.div(tf.reduce_sum(loss * fpmask), batch_size, name='value')".  batch_size maybe modified as anchor number.

**Comment by muou55555** [comment]
I think the point is  "loss = tf.div(tf.reduce_sum(loss * fpmask), batch_size, name='value')". batch_size maybe modified as anchor number.

**Comment by vladtamas** [comment]
@muou55555 Did you find the solution? I have a similar issue

**Comment by qingchengboy** [comment]
@muou55555 My problem is also this

**Comment by AriesJin** [comment]
i have the similar problem. anyone know why？

**Comment by jhgfkdj** [comment]
I got this problem as well , anybody solved it ?
